src/python/tensor.py

```diff
@@ -0,0 +1,292 @@
+import sys, os
+import numpy as np
+import singa
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../'))
+from core_pb2 import *
+
+
+''' Class and member functions for singa::Tensor
+'''
+class Tensor(object):
+
+  def __init__(self, arg=None, device=None, dtype=kFloat32):
+    ''' arg = Tensor or tuple for shape
```

Comment:
may not need to use arg, just use shape.
"Assignment statements in Python do not copy objects, they create bindings between a target and an object"
e.g., Tensor A = B. A is then a reference of B. 

We may add __copy__ for shallow copy which copies the meta data and share the internal data;  and __deepcopy__ for deep copy which calls CopyDataToFrom to copy internal data and meta data.